### PR TITLE
Setting up methods for event handling and state change

### PR DESCRIPTION
**Pull request by jsharples777**
These need implementation in App.js

Added a bunch of methods as placeholders for the event calls as follows
handleEventStartRecipeSearch

handleEventAddRecipeToFavourites
handleEventRemoveRecipeFromFavourites 

handleEventAddRecipeToShoppingList
handleEventRemoveIngredientFromShoppingList 

handleEventShowShoppingList 
handleEventShowFavouriteRecipes 
handleEventShowRecipeDetails

### CODE REVIEW

**Review by jsharples777 — COMMENTED** [review]
Added event handling methods for the various user events

**Review by jsharples777 — COMMENTED** [review]
Added search by recipeId to support events
If the event.targets have an attribute storing the event id, it's easy to find the recipe object from the state manager

File: public/js/App.js [hunk 1]
```diff
@@ -33,37 +52,192 @@ class App {
...
 var favBtn = document.querySelector("#favourite-btn"); // the button to open the favourites modal
 var shoppingListBtn = document.querySelector("#shopping-list-btn"); // the button to open the shopping list modal
 var supermarketButton = document.querySelector("#supermarket-btn"); // the button to open the supermarket map feature, may not need this
 var resultsList = document.querySelector("#results-list"); // the container that we will append our search results elements to
 var recipeDetail = document.querySelector("#recipe-detail"); // the modal window for recipe detail
+
+
+searchBtn.addEventListener("click",app.handleEventStartRecipeSearch);
+shoppingListBtn.addEventListener("click",app.handleEventShowShoppingList);
+favBtn.addEventListener("click",app.handleEventShowFavouriteRecipes);
```
Comment (raven-bootcamp) [inline public/js/App.js]:
```suggestion
searchBtn.addEventListener("click", app.handleEventStartRecipeSearch);
shoppingListBtn.addEventListener("click", app.handleEventShowShoppingList);
favBtn.addEventListener("click", app.handleEventShowFavouriteRecipes);
```

**Review by raven-bootcamp — CHANGES_REQUESTED** [review]
Looks good besides one small paragraph that needs some spaces 😄

**Review by AMIrene — APPROVED** [review]

**Review by AMIrene — APPROVED** [review]

**Review by AMIrene — APPROVED** [review]

**Review by AMIrene — APPROVED** [review]

**Review by raven-bootcamp — APPROVED** [review]

**Pull request merged**